<!DOCTYPE html>
<html lang="en"><head>
<script src="HomoeologousExpressionAnalysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="HomoeologousExpressionAnalysis_files/libs/quarto-html/tabby.min.js"></script>
<script src="HomoeologousExpressionAnalysis_files/libs/quarto-html/popper.min.js"></script>
<script src="HomoeologousExpressionAnalysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="HomoeologousExpressionAnalysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="HomoeologousExpressionAnalysis_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="HomoeologousExpressionAnalysis_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="HomoeologousExpressionAnalysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.433">

  <meta name="author" content="Moeko Okada">
  <title>Homoeologous Expression Analysis</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="HomoeologousExpressionAnalysis_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="HomoeologousExpressionAnalysis_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="HomoeologousExpressionAnalysis_files/libs/revealjs/dist/theme/quarto.css">
  <link href="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Homoeologous Expression Analysis</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Moeko Okada 
</div>
</div>
</div>

</section>
<section id="commands-scripts-are-available-on-github" class="slide level2">
<h2>Commands/ scripts are available on GitHub</h2>
<p><a href="https://github.com/MoekoOkada/SummerSchool_2023">https://github.com/MoekoOkada/SummerSchool_2023</a></p>
<p>! When you copy the scripts, please copy from the GitHub page, not from this HTML file!</p>
</section>
<section id="todays-plan" class="slide level2">
<h2>Today’s Plan</h2>
<div style="font-size: 0.8em">
<ol type="1">
<li>Data download</li>
<li>Quality control</li>
<li>Homoeologous Mapping</li>
<li>Read sorting by EAGLE-RC</li>
<li>Read count
<ol type="1">
<li>read count</li>
<li>make a count matrix</li>
</ol></li>
<li>Expression analysis
<ol type="1">
<li>Differential expression analysis</li>
<li>Clustering</li>
<li>Homoeolog ratio test</li>
</ol></li>
</ol>
</div>
</section>
<section id="rna-seq-workflow-review" class="slide level2">
<h2>RNA-seq workflow review</h2>
<ul>
<li>Sequencing</li>
<li>Quality control</li>
<li>Mapping/ Alignment</li>
<li>Read count</li>
<li>Estimate Expression level
<ul>
<li>Differential expression analysis</li>
<li>Functional Analyses</li>
<li>etc…</li>
</ul></li>
</ul>
</section>
<section id="expression-analysis-in-polyploids" class="slide level2">
<h2>Expression analysis in polyploids</h2>

<img data-src="./photos/HomoeologousExpression.jpeg" class="r-stretch quarto-figure-center"><p class="caption">KK Shimizu (2022) Curr. Opin. Plant Biol.</p></section>
<section id="rna-seq-workflow-review-1" class="slide level2">
<h2>RNA-seq workflow review</h2>
<ul>
<li>Sequencing</li>
<li>Quality control</li>
<li><strong>Homoeologous Mapping/ Alignment</strong></li>
<li><strong>Read classification</strong></li>
<li>Read count</li>
<li>Estimate Expression level
<ul>
<li>Differential expression analysis</li>
<li>Functional Analyses</li>
<li>etc…</li>
</ul></li>
</ul>
</section>
<section id="what-you-need-to-use-eagle-rc" class="slide level2">
<h2>What you need to use EAGLE-RC</h2>
<ul>
<li>Reference genome sequence</li>
<li>NGS reads (DNA-seq, RNA-seq, methylome)</li>
<li>Gene annotation file (gtf, gff)</li>
</ul>
</section>
<section id="todays-sample" class="slide level2">
<h2>Today’s sample</h2>
<p><strong><em>Arabidopsis kamchatica</em></strong></p>
<p><br></p>
<ul>
<li>Accession: murodo</li>
<li>RNA from root</li>
<li>Control and zinc treatment for 48h</li>
<li>3 replicates for each condition</li>
</ul>
<p><img data-src="./photos/kam.jpg" class="absolute" style="top: 50px; right: 50px; width: 30%; "></p>

<aside><div>
<p>https://www.bg.s.u-tokyo.ac.jp/nikko-old/5_jokyo/species/Arabidopsis_kamchatica.html</p>
</div></aside></section>
<section id="todays-plan-1" class="slide level2">
<h2>Today’s Plan</h2>
<div style="font-size: 0.8em">
<ol type="1">
<li>Data download</li>
<li>Quality control</li>
<li>Homoeologous Mapping</li>
<li>Read sorting by EAGLE-RC</li>
<li>Read count
<ol type="1">
<li>read count</li>
<li>make a count matrix</li>
</ol></li>
<li>Expression analysis
<ol type="1">
<li>Differential expression analysis</li>
<li>Clustering</li>
<li>Homoeolog ratio test</li>
</ol></li>
</ol>
</div>
</section>
<section id="jfyi" class="slide level2">
<h2>JFYI…</h2>
<div style="font-size: 0.8em">
<ul>
<li>Lines with <code>$</code> mean commands you use.
<ul>
<li>If you want to copy and paste, exclude <code>$</code>.</li>
</ul></li>
<li>I put the name of the directory you should be before <code>$</code>.
<ul>
<li>If you are not there, please run the <code>cd</code> command</li>
</ul></li>
<li>Lines without <code>$</code> mean standard output on the terminal.</li>
</ul>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="ex">$</span> cd src</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="ex">src</span> $ pwd</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="ex">/tmp/eagle/work/polyploid-rna-seq-analyses/src</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="ex">src</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="data-download" class="slide level2">
<h2>1. Data download</h2>
<p>Error messages may appear during extraction, but just ignore them!</p>
<p><br></p>
<h4 id="download-genome-data">1-1. Download genome data</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">$</span> cd src</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="ex">src</span> $ wget <span class="at">-O</span> genome.tar.gz https://drive.switch.ch/index.php/s/ZQzcTZ5lGJEcCbA/download</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="ex">src</span> $ ls</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="ex">genome.tar.gz</span>  scripts</span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="ex">src</span> $ tar xvzf genome.tar.gz</span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="ex">src</span> $ ls</span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="ex">genome</span>  genome.tar.gz  scripts</span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="ex">src</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="data-download-1" class="slide level2">
<h2>1. Data download</h2>
<p><br></p>
<h4 id="download-raw-sequence-data">1-2. Download raw sequence data</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="ex">src</span> $ wget <span class="at">-O</span> fastq.tar.gz https://drive.switch.ch/index.php/s/0ny11xweoA5WhEX/download</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="ex">src</span> $ ls</span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="ex">fastq.tar.gz</span>  genome  genome.tar.gz  scripts</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="ex">src</span> $ tar xvzf fastq.tar.gz</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="ex">src</span> $ ls</span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="ex">fastq</span>  fastq.tar.gz  genome  genome.tar.gz  scripts</span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="ex">src</span> $ ls fastq</span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="ex">MUR_1_R1_ds.fastq.gz</span>  MUR_3_R1_ds.fastq.gz       MUR_R48h_2_R1_ds.fastq.gz  trimmomatic_adapters.txt</span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="ex">MUR_2_R1_ds.fastq.gz</span>  MUR_R48h_1_R1_ds.fastq.gz  MUR_R48h_3_R1_ds.fastq.gz</span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="ex">src</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="quality-control" class="slide level2">
<h2>2. Quality control</h2>
<p>Trimmomatic (<a href="https://academic.oup.com/bioinformatics/article/30/15/2114/2390096%3E">Bolger et al.&nbsp;<em>Bioinformatics</em> 2014.</a>)</p>
<ul>
<li>Remove sequencer-specific adapter sequence</li>
<li>Remove low-quality reads (Q &lt; 30)</li>
<li>Remove too short reads (&lt; 40)</li>
<li>Use only reads passed the criteria</li>
</ul>
</section>
<section id="quality-control-1" class="slide level2">
<h2>2. Quality control</h2>
<p>Use shell script <code>1_qc.sh</code> to run Trimmomatics for all data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode numberSource shell number-lines code-with-copy"><code class="sourceCode"><span id="cb4-1"><a href="#cb4-1"></a>#!/bin/sh</span>
<span id="cb4-2"><a href="#cb4-2"></a># 1_qc.sh</span>
<span id="cb4-3"><a href="#cb4-3"></a># Project: SummerSchool</span>
<span id="cb4-4"><a href="#cb4-4"></a># Date: 4-Jul-2023</span>
<span id="cb4-5"><a href="#cb4-5"></a># Author: Moeko Okada</span>
<span id="cb4-6"><a href="#cb4-6"></a></span>
<span id="cb4-7"><a href="#cb4-7"></a>mkdir 1_qc</span>
<span id="cb4-8"><a href="#cb4-8"></a>cd fastq/</span>
<span id="cb4-9"><a href="#cb4-9"></a></span>
<span id="cb4-10"><a href="#cb4-10"></a>for i in `ls *_R1_ds.fastq.gz`; do</span>
<span id="cb4-11"><a href="#cb4-11"></a>  f=`basename ${i} _R1_ds.fastq.gz`</span>
<span id="cb4-12"><a href="#cb4-12"></a>  echo "# start ${f} " $(date)</span>
<span id="cb4-13"><a href="#cb4-13"></a></span>
<span id="cb4-14"><a href="#cb4-14"></a>  trimmomatic SE -phred33 -threads 2 \</span>
<span id="cb4-15"><a href="#cb4-15"></a>    ${i} ../1_qc/${f}_ds_clean.fastq.gz \</span>
<span id="cb4-16"><a href="#cb4-16"></a>    ILLUMINACLIP:trimmomatic_adapters.txt:2:30:10 \</span>
<span id="cb4-17"><a href="#cb4-17"></a>    SLIDINGWINDOW:4:30 MINLEN:40</span>
<span id="cb4-18"><a href="#cb4-18"></a></span>
<span id="cb4-19"><a href="#cb4-19"></a>  echo "# done ${f} " $(date)</span>
<span id="cb4-20"><a href="#cb4-20"></a>done</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="quality-control-2" class="slide level2">
<h2>2. Quality control</h2>
<p>Make sure you are in the <code>src</code> directory.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="ex">src</span> $ bash scripts/1_qc.sh</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="ex">src</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br></p>
<div style="font-size : 0.8em">
<p>You will get 4 <code>.fastq.gz</code> files in <code>src/1_qc</code></p>
</div>
</section>
<section id="homoeologous-mapping" class="slide level2">
<h2>3. Homoeologous Mapping</h2>
<div style="font-size:0.8em">
<p>For read classification by EAGLE-RC, reads should be mapped to each subgenome respectively.</p>
</div>
<p><img data-src="./photos/sorting.png" class="absolute" style="right: 300px; height: 60%; "></p>

<aside><div>
<p>Kuo et al.&nbsp;<em>Brief. Bioinfom</em> 2020</p>
</div></aside></section>
<section id="mapping-software" class="slide level2">
<h2>Mapping software</h2>
<div style="font-size:0.7em">
<ul>
<li>BWA (DNA reads to genome sequence) (<a href="https://academic.oup.com/bioinformatics/article/25/14/1754/225615">Li and Durbin <em>Bioinformatics</em> 2009.</a>)</li>
<li><strong>HISAT2</strong> (RNA/DNA reads to genome sequence) (<a href="https://www.nature.com/articles/s41587-019-0201-4">Kim et al.&nbsp;<em>Nat. Biotech.</em> 2019.</a>)</li>
<li>bowtie2 (RNA reads to transcriptome sequence) (<a href="https://www.nature.com/articles/nmeth.3317">Kim et al.&nbsp;<em>Nat. Method</em> 2015</a>)</li>
<li>STAR (DNA/RNA to genome sequence) (<a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537">Dobin et al.&nbsp;<em>Bioinformatics</em> 2013</a>)</li>
</ul>
<p>STAR is recommended to EAGLE-RC.</p>
<p>But today we will use HISAT2 because of the time and resource limitations.</p>
<p><br></p>
<h6 id="other-required-software">Other required software</h6>
<p>samtools (<a href="https://academic.oup.com/bioinformatics/article/25/16/2078/204688">Li et al.&nbsp;<em>Bioinfirmatics</em> 2008.</a>)</p>
</div>
</section>
<section id="homoeologous-mapping-1" class="slide level2">
<h2>3. Homoeologous Mapping</h2>
<p>Use shell script <code>2_map.sh</code> to run Trimmomatics for all data.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode numberSource shell number-lines code-with-copy"><code class="sourceCode"><span id="cb6-1"><a href="#cb6-1"></a>#!/bin/sh</span>
<span id="cb6-2"><a href="#cb6-2"></a># 2_map.sh</span>
<span id="cb6-3"><a href="#cb6-3"></a># Project: SummerSchool</span>
<span id="cb6-4"><a href="#cb6-4"></a># Date: 4-Jul-2023</span>
<span id="cb6-5"><a href="#cb6-5"></a># Author: Moeko Okada</span>
<span id="cb6-6"><a href="#cb6-6"></a></span>
<span id="cb6-7"><a href="#cb6-7"></a>mkdir 2_map</span>
<span id="cb6-8"><a href="#cb6-8"></a>cd 1_qc/</span>
<span id="cb6-9"><a href="#cb6-9"></a></span>
<span id="cb6-10"><a href="#cb6-10"></a>for i in `ls *_ds_clean.fastq.gz`; do</span>
<span id="cb6-11"><a href="#cb6-11"></a>  f=`basename ${i} _clean.fastq.gz`</span>
<span id="cb6-12"><a href="#cb6-12"></a>  echo "# start ${f} " $(date)</span>
<span id="cb6-13"><a href="#cb6-13"></a></span>
<span id="cb6-14"><a href="#cb6-14"></a>  hisat2 -p 2 -x ../genome/Ahal_v2_2 -U ${i} | samtools view -SbF4 -q 40 - | samtools sort -m 8G -@ 2 -o ../2_map/${f}_hal.bam</span>
<span id="cb6-15"><a href="#cb6-15"></a>  hisat2 -p 2 -x ../genome/Alyr_v2_2 -U ${i} | samtools view -SbF4 -q 40 - | samtools sort -m 8G -@ 2 -o ../2_map/${f}_lyr.bam</span>
<span id="cb6-16"><a href="#cb6-16"></a>  samtools index ../2_map/${f}_hal.bam</span>
<span id="cb6-17"><a href="#cb6-17"></a>  samtools index ../2_map/${f}_lyr.bam</span>
<span id="cb6-18"><a href="#cb6-18"></a></span>
<span id="cb6-19"><a href="#cb6-19"></a>  echo "# done ${f} " $(date)</span>
<span id="cb6-20"><a href="#cb6-20"></a>done</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="homoeologous-mapping-2" class="slide level2">
<h2>3. Homoeologous Mapping</h2>
<p>Make sure you are in the <code>src</code> directory.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode numberSource shell number-lines code-with-copy"><code class="sourceCode"><span id="cb7-1"><a href="#cb7-1"></a>src $ bash scripts/2_map.sh</span>
<span id="cb7-2"><a href="#cb7-2"></a>src $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br></p>
<div style="font-size : 0.8em">
<p>You will get 4 <code>.bam</code> and 4 <code>.bam.bai</code> files in <code>src/2_map</code></p>
</div>
</section>
<section id="read-sorting-by-eagle-rc" class="slide level2">
<h2>4. Read sorting by EAGLE-RC</h2>
<div style="font-size:0.8em">
<p>EAGLE-RC (<a href="https://academic.oup.com/bib/article/21/2/395/5251019">Kuo et al.&nbsp;<em>Brief. Bioinfom</em> 2020</a>)</p>
<ul>
<li>Normal mode
<ul>
<li>Requires gene annotation and homoeologous list</li>
<li>High accuracy</li>
<li>Takes a while</li>
</ul></li>
<li>NGI mode
<ul>
<li>Requires only reference genome info (fasta and .fai)</li>
<li>Low accuracy</li>
<li>Fast</li>
</ul></li>
</ul>
</div>

<aside><div>
<p>https://github.com/tony-kuo/eagle</p>
</div></aside></section>
<section id="read-sorting-by-eagle-rc-1" class="slide level2">
<h2>4. Read sorting by EAGLE-RC</h2>
<div style="font-size:0.9em">
<p>Today, we will use NGI mode.</p>
<p><br></p>
<p>Please take a look <a href="https://github.com/tony-kuo/eagle">github page</a> for</p>
<ul>
<li>creating a homoeologous list</li>
<li>how to run normal mode</li>
<li>how to use eagle-rc for genotyping from DNA-seq data</li>
<li>etc…</li>
</ul>
<p>You may be required to install some software.</p>
</div>

<aside><div>
<p>https://github.com/tony-kuo/eagle</p>
</div></aside></section>
<section id="read-sorting-by-eagle-rc-2" class="slide level2">
<h2>4. Read sorting by EAGLE-RC</h2>
<p>Use shell script <code>3_eagle.sh</code> to run Trimmomatics for all data.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode numberSource shell number-lines code-with-copy"><code class="sourceCode"><span id="cb8-1"><a href="#cb8-1"></a>#!/bin/sh</span>
<span id="cb8-2"><a href="#cb8-2"></a># 3_eagle.sh</span>
<span id="cb8-3"><a href="#cb8-3"></a># Project: SummerSchool</span>
<span id="cb8-4"><a href="#cb8-4"></a># Date: 4-Jul-2023</span>
<span id="cb8-5"><a href="#cb8-5"></a># Author: Moeko Okada</span>
<span id="cb8-6"><a href="#cb8-6"></a></span>
<span id="cb8-7"><a href="#cb8-7"></a>mkdir 3_eagle</span>
<span id="cb8-8"><a href="#cb8-8"></a>cd 2_map/</span>
<span id="cb8-9"><a href="#cb8-9"></a></span>
<span id="cb8-10"><a href="#cb8-10"></a>for i in `ls *_hal.bam`; do</span>
<span id="cb8-11"><a href="#cb8-11"></a>  f=`basename ${i} _hal.bam`</span>
<span id="cb8-12"><a href="#cb8-12"></a>  echo "# start ${f} " $(date)</span>
<span id="cb8-13"><a href="#cb8-13"></a></span>
<span id="cb8-14"><a href="#cb8-14"></a>  eagle-rc --ngi --ref1=../genome/Ahal_v2_2.fa --ref2=../genome/Alyr_v2_2.fa --bam1=../2_map/${i} --bam2=../2_map/${f}_lyr.bam -o ../3_eagle/${f}.eaglerc &gt; ../3_eagle/${f}.eaglerc.stdout.log 2&gt; ../3_eagle/${f}.eaglerc.errout.log</span>
<span id="cb8-15"><a href="#cb8-15"></a></span>
<span id="cb8-16"><a href="#cb8-16"></a>  echo "# done ${f} " $(date)</span>
<span id="cb8-17"><a href="#cb8-17"></a>done</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="read-sorting-by-eagle-rc-3" class="slide level2">
<h2>4. Read sorting by EAGLE-RC</h2>
<p>Make sure you are in the <code>src</code> directory!</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1"></a><span class="ex">src</span> $ bash scripts/3_eagle.sh</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="ex">src</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="output-file-of-eagle-rc" class="slide level2">
<h2>Output file of EAGLE-RC</h2>
<div style="font-size:0.8em">
<ul>
<li>*.ref.bam: reads with the reference allele</li>
<li>*.alt.bam: reads with the alternative allele</li>
<li>*.unk.bam: ambiguous reads</li>
<li>*.mul.bam: multi-allele reads (empty when we use NGI mode)</li>
</ul>
<p><br></p>
<ul>
<li>eaglerc1: based on reference 1 (today’s case: halleri genome)</li>
<li>eaglerc2: based on reference 2 (today’s case: lyrata genome)</li>
</ul>
<p>To make it faster and simpler, we will use <code>*.eaglerc1.ref.bam</code> and <code>*.eaglerc1.alt.bam</code>.</p>
</div>
</section>
<section id="count-mapped-read" class="slide level2">
<h2>5-1. Count mapped read</h2>
<p>featureCounts (<a href="https://academic.oup.com/bioinformatics/article/30/7/923/232889">Liao et al.&nbsp;<em>Bioinformatics</em> 2014.</a>)</p>
<p><br></p>
<h4 id="requirements">Requirements</h4>
<ul>
<li>gene annotation (gtf file)</li>
<li>mapped bam file</li>
</ul>
</section>
<section id="count-mapped-read-1" class="slide level2">
<h2>5-1. Count mapped read</h2>
<p>Use shell script <code>4_count.sh</code> to run Trimmomatics for all data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode numberSource shell number-lines code-with-copy"><code class="sourceCode"><span id="cb10-1"><a href="#cb10-1"></a>#!/bin/sh</span>
<span id="cb10-2"><a href="#cb10-2"></a># 4_count.sh</span>
<span id="cb10-3"><a href="#cb10-3"></a># Project: SummerSchool</span>
<span id="cb10-4"><a href="#cb10-4"></a># Date: 4-Jul-2023</span>
<span id="cb10-5"><a href="#cb10-5"></a># Author: Moeko Okada</span>
<span id="cb10-6"><a href="#cb10-6"></a></span>
<span id="cb10-7"><a href="#cb10-7"></a>mkdir 4_count</span>
<span id="cb10-8"><a href="#cb10-8"></a>cd 3_eagle/</span>
<span id="cb10-9"><a href="#cb10-9"></a></span>
<span id="cb10-10"><a href="#cb10-10"></a>for i in `ls *.eaglerc1.ref.bam`; do</span>
<span id="cb10-11"><a href="#cb10-11"></a>  f=`basename ${i} .eaglerc1.ref.bam`</span>
<span id="cb10-12"><a href="#cb10-12"></a>  echo "# start ${f} " $(date)</span>
<span id="cb10-13"><a href="#cb10-13"></a></span>
<span id="cb10-14"><a href="#cb10-14"></a>  featureCounts -T 2 -t exon -g transcript_id -a ../genome/Ahal_v2_2.gtf -o ../4_count/${f}_hal_counts.txt ${i}</span>
<span id="cb10-15"><a href="#cb10-15"></a>  featureCounts -T 2 -t exon -g transcript_id -a ../genome/Alyr_v2_2.gtf -o ../4_count/${f}_lyr_counts.txt ${f}.eaglerc2.ref.bam</span>
<span id="cb10-16"><a href="#cb10-16"></a></span>
<span id="cb10-17"><a href="#cb10-17"></a>  echo "# done ${f} " $(date)</span>
<span id="cb10-18"><a href="#cb10-18"></a>done</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="count-mapped-read-2" class="slide level2">
<h2>5-1. Count mapped read</h2>
<p>Make sure you are in the “src” directory.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1"></a><span class="ex">src</span> $ bash scripts/4_count.sh</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="ex">src</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="make-read-count-matrix" class="slide level2">
<h2>5-2. Make read count matrix</h2>
<p>For further expression analysis, prepare a count matrix that looks like this.</p>
<div style="font-size:0.8em">
<table>
<thead>
<tr class="header">
<th>gene_id</th>
<th style="text-align: right;">MUR_1</th>
<th style="text-align: right;">MUR_2</th>
<th style="text-align: right;">MUR_3</th>
<th style="text-align: right;">MUR_1_48h</th>
<th style="text-align: right;">MUR_2_48h</th>
<th style="text-align: right;">MUR_3_48h</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>gene1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">58</td>
<td style="text-align: right;">62</td>
<td style="text-align: right;">65</td>
</tr>
<tr class="even">
<td>gene2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td>gene3</td>
<td style="text-align: right;">100</td>
<td style="text-align: right;">103</td>
<td style="text-align: right;">99</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">6</td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="make-expression-table" class="slide level2">
<h2>5-2. Make Expression table</h2>
<p>Make sure you are in the <code>4_count</code> directory!</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1"></a><span class="ex">src</span> $ cd 4_count</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="ex">4_count</span> $ <span class="er">(</span><span class="bu">echo</span> <span class="st">"gene_id"</span><span class="kw">;</span><span class="fu">ls</span> <span class="pp">*</span>_hal_counts.txt<span class="kw">;</span> <span class="kw">)</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-e</span> s/_hal_counts.txt//g <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'\n'</span> <span class="st">'_hal\t'</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/\s*$//'</span> <span class="op">&gt;</span> hal_counts.tsv</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="ex">4_count</span> $ <span class="er">(</span><span class="bu">echo</span> <span class="st">"gene_id"</span><span class="kw">;</span><span class="fu">ls</span> <span class="pp">*</span>_lyr_counts.txt<span class="kw">;</span> <span class="kw">)</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-e</span> s/_ds_lyr_counts.txt//g <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'\n'</span> <span class="st">'_lyr\t'</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/\s*$//'</span> <span class="op">&gt;</span> lyr_counts.tsv</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="ex">4_count</span> $ echo <span class="st">""</span> <span class="op">&gt;&gt;</span> hal_counts.tsv</span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="ex">4_count</span> $ echo <span class="st">""</span> <span class="op">&gt;&gt;</span> lyr_counts.tsv</span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="ex">4_count</span> $ python /tmp/eagle/scripts/tablize.py <span class="at">-skip</span> 1 <span class="at">-a</span> <span class="at">-i</span> 0 <span class="at">-c</span> 6 <span class="pp">*</span>_hal_counts.txt <span class="op">&gt;&gt;</span> hal_counts.tsv</span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="ex">4_count</span> $ python /tmp/eagle/scripts/tablize.py <span class="at">-skip</span> 1 <span class="at">-a</span> <span class="at">-i</span> 0 <span class="at">-c</span> 6 <span class="pp">*</span>_lyr_counts.txt <span class="op">&gt;&gt;</span> lyr_counts.tsv</span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="ex">4_count</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br></p>
</section>
<section id="expression-analysis" class="slide level2">
<h2>6-1. Expression analysis</h2>
<p>Use R for calculation and visualization.</p>
<p>Please move to the R console.</p>

<img data-src="./photos/console.png" class="r-stretch"><p><br></p>
<p>Required package information is written in <a href="https://github.com/MoekoOkada/SummerSchool_2023/blob/main/scripts/install.r">install.R</a>.</p>
</section>
<section id="required-libraries" class="slide level2">
<h2>Required libraries</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">library</span>(edgeR) <span class="co"># normalisation, deg detection</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="fu">library</span>(DESeq2) <span class="co"># normalisation, deg detection</span></span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="fu">library</span>(dplyr) <span class="co"># data formatting</span></span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="fu">library</span>(stringr) <span class="co"># data formatting</span></span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="fu">library</span>(gplots) <span class="co"># plot heatmap</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="make-expression-table-1" class="slide level2">
<h2>6-1-1. Make Expression Table</h2>
<p><br></p>
<h3 id="load-data">Load data</h3>
<div class="sourceCode" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="do">## load count data</span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="co"># set working directory</span></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="fu">setwd</span>(<span class="st">"src/exp"</span>)</span>
<span id="cb14-4"><a href="#cb14-4"></a></span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="co"># load count data of halleri side</span></span>
<span id="cb14-6"><a href="#cb14-6"></a>hal_count <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"hal_counts.tsv"</span>, <span class="at">header =</span> T, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb14-7"><a href="#cb14-7"></a></span>
<span id="cb14-8"><a href="#cb14-8"></a><span class="co"># check hal count data</span></span>
<span id="cb14-9"><a href="#cb14-9"></a><span class="fu">head</span>(hal_count)</span>
<span id="cb14-10"><a href="#cb14-10"></a></span>
<span id="cb14-11"><a href="#cb14-11"></a><span class="co"># load count data of lyrata side</span></span>
<span id="cb14-12"><a href="#cb14-12"></a>lyr_count <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"lyr_counts.tsv"</span>, <span class="at">header =</span> T, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb14-13"><a href="#cb14-13"></a></span>
<span id="cb14-14"><a href="#cb14-14"></a><span class="co"># check lyr count data</span></span>
<span id="cb14-15"><a href="#cb14-15"></a><span class="fu">head</span>(lyr_count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="data-prep.-for-eagle-rc" class="slide level2">
<h2>6-1-2. Data prep. for EAGLE-RC</h2>
<ol type="1">
<li>Make a whole count matrix</li>
<li>Load the gene length info of halleri</li>
<li>calculate RPKM using DESeq2</li>
</ol>
</section>
<section id="data-prep.-for-eagle-rc-1" class="slide level2">
<h2>6-2. Data prep. for EAGLE-RC</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co"># bind count data</span></span>
<span id="cb15-2"><a href="#cb15-2"></a>count <span class="ot">&lt;-</span> <span class="fu">cbind</span>(hal_count, lyr_count) <span class="co"># cbind merges the two data frame based on rownames</span></span>
<span id="cb15-3"><a href="#cb15-3"></a><span class="co"># check data</span></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="fu">head</span>(count)</span>
<span id="cb15-5"><a href="#cb15-5"></a></span>
<span id="cb15-6"><a href="#cb15-6"></a><span class="co"># load gene length of hal</span></span>
<span id="cb15-7"><a href="#cb15-7"></a>Ahal_gtf <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../genome/Ahal_v2_2.gtf"</span>, <span class="at">header =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb15-8"><a href="#cb15-8"></a>Ahal_gtf <span class="ot">&lt;-</span> <span class="fu">filter</span>(Ahal_gtf, V3 <span class="sc">==</span> <span class="st">"transcript"</span>) <span class="co"># extract transcript information</span></span>
<span id="cb15-9"><a href="#cb15-9"></a>len <span class="ot">&lt;-</span> Ahal_gtf<span class="sc">$</span>V5 <span class="sc">-</span> Ahal_gtf<span class="sc">$</span>V4 <span class="sc">+</span> <span class="dv">1</span> <span class="co"># define length based on start and end position of each gene</span></span>
<span id="cb15-10"><a href="#cb15-10"></a></span>
<span id="cb15-11"><a href="#cb15-11"></a><span class="co"># Calculate RPKM</span></span>
<span id="cb15-12"><a href="#cb15-12"></a>rpkm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rpkm</span>(count, len, <span class="at">log =</span> <span class="cn">FALSE</span>)) <span class="co"># rpkm function in DESeq2 package</span></span>
<span id="cb15-13"><a href="#cb15-13"></a>table_rpkm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(rpkm, <span class="at">n =</span> <span class="fu">nrow</span>(count)) <span class="co"># convert to table</span></span>
<span id="cb15-14"><a href="#cb15-14"></a><span class="fu">head</span>(table_rpkm)</span>
<span id="cb15-15"><a href="#cb15-15"></a><span class="fu">write.table</span>(table_rpkm, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"homoeolog_RPKM.txt"</span>), <span class="at">col.names =</span> T, <span class="at">row.names =</span> T, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="co"># Output</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="make-a-count-matrix" class="slide level2">
<h2>6-1-3. Make a count matrix</h2>
<p><br></p>
<p>Make a count matrix for each condition (control and zinc) for expression analysis</p>
<p><br></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># Select function in dplyr package</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>control <span class="ot">&lt;-</span> <span class="fu">select</span>(count, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"48h"</span>)) <span class="co"># Select columns without "48h"</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="fu">head</span>(control)</span>
<span id="cb16-4"><a href="#cb16-4"></a>zinc <span class="ot">&lt;-</span> <span class="fu">select</span>(count, <span class="fu">contains</span>(<span class="st">"48h"</span>)) <span class="co"># Select columns with "48h"</span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="fu">head</span>(zinc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="expression-analysis-1" class="slide level2">
<h2>6-1-4. Expression analysis</h2>
<p>Use edgeR to compare halleri side and kamchatica side under zinc treatment.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="co"># set groups</span></span>
<span id="cb17-2"><a href="#cb17-2"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">"kam_hal"</span>, <span class="st">"kam_hal"</span>, <span class="st">"kam_hal"</span>, <span class="st">"kam_lyr"</span>, <span class="st">"kam_lyr"</span>, <span class="st">"kam_lyr"</span>))</span>
<span id="cb17-3"><a href="#cb17-3"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group)</span>
<span id="cb17-4"><a href="#cb17-4"></a></span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="co"># DEG detection from zinc-treated samples</span></span>
<span id="cb17-6"><a href="#cb17-6"></a>d1 <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> zinc, <span class="at">group =</span> group)</span>
<span id="cb17-7"><a href="#cb17-7"></a></span>
<span id="cb17-8"><a href="#cb17-8"></a><span class="co"># normalize according to UserGuide</span></span>
<span id="cb17-9"><a href="#cb17-9"></a>d1 <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(d1)</span>
<span id="cb17-10"><a href="#cb17-10"></a>d1 <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(d1, design)</span>
<span id="cb17-11"><a href="#cb17-11"></a></span>
<span id="cb17-12"><a href="#cb17-12"></a><span class="co"># plot Multi-dimensional scaling</span></span>
<span id="cb17-13"><a href="#cb17-13"></a><span class="fu">plotMDS</span>(d1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="differential-expression-analysis" class="slide level2">
<h2>6-1-5. Differential Expression Analysis</h2>
<h3 id="msd-plot">MSD plot</h3>

<img data-src="./photos/mds.png" class="r-stretch"></section>
<section id="clustering-and-heatmap" class="slide level2">
<h2>6-2. Clustering and heatmap</h2>
<p><br></p>
<h3 id="format-data-to-draw-a-heatmap.">6-2-1. format data to draw a heatmap.</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># calculate cpm and format it for heatmap</span></span>
<span id="cb18-2"><a href="#cb18-2"></a>d2 <span class="ot">&lt;-</span> <span class="fu">cpm</span>(d1, <span class="at">normalized.lib.size =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-3"><a href="#cb18-3"></a>scaledata <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(d2))) <span class="co"># Centers and scales data.</span></span>
<span id="cb18-4"><a href="#cb18-4"></a>scaledata <span class="ot">&lt;-</span> scaledata[<span class="fu">complete.cases</span>(scaledata), ]</span>
<span id="cb18-5"><a href="#cb18-5"></a></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="co"># Change ids</span></span>
<span id="cb18-7"><a href="#cb18-7"></a><span class="fu">rownames</span>(scaledata) <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">rownames</span>(scaledata), gsub, <span class="at">pattern =</span> <span class="st">".t1"</span>, <span class="at">replacement =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="extract-metal-related-genes-from-the-expression-table." class="slide level2">
<h2>6-2-2. Extract metal-related genes from the expression table.</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># make a gene set</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>metal_genes <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"metal_genes_ids.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> T)</span>
<span id="cb19-3"><a href="#cb19-3"></a>metal_genes[, <span class="dv">4</span>]</span>
<span id="cb19-4"><a href="#cb19-4"></a></span>
<span id="cb19-5"><a href="#cb19-5"></a><span class="co"># Extract metal genes from deg files</span></span>
<span id="cb19-6"><a href="#cb19-6"></a>my.geneset <span class="ot">&lt;-</span> scaledata[<span class="fu">rownames</span>(scaledata) <span class="sc">%in%</span> metal_genes[, <span class="dv">4</span>], ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="hierarchical-clustering" class="slide level2">
<h2>6-2-3. Hierarchical clustering</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="co"># clustering</span></span>
<span id="cb20-2"><a href="#cb20-2"></a>hr <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(<span class="fu">t</span>(my.geneset), <span class="at">method =</span> <span class="st">"pearson"</span>)), <span class="at">method =</span> <span class="st">"complete"</span>)</span>
<span id="cb20-3"><a href="#cb20-3"></a>hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(my.geneset, <span class="at">method =</span> <span class="st">"spearman"</span>)), <span class="at">method =</span> <span class="st">"complete"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="plot-heatmap-based-on-the-clustering-information." class="slide level2">
<h2>6-2-4. Plot heatmap based on the clustering information.</h2>
<p>Use the <code>heatmap.2</code> function in the <code>gplot</code> package.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="co"># Clustering of metal-related genes</span></span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="fu">png</span>(<span class="fu">paste0</span>(<span class="st">"Cluster_hm_zinc.png"</span>))</span>
<span id="cb21-3"><a href="#cb21-3"></a><span class="fu">heatmap.2</span>(my.geneset,</span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="at">Rowv =</span> <span class="fu">as.dendrogram</span>(hr),</span>
<span id="cb21-5"><a href="#cb21-5"></a>  <span class="at">Colv =</span> <span class="fu">as.dendrogram</span>(hc),</span>
<span id="cb21-6"><a href="#cb21-6"></a>  <span class="at">col =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"yellow"</span>))(<span class="dv">100</span>),</span>
<span id="cb21-7"><a href="#cb21-7"></a>  <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb21-8"><a href="#cb21-8"></a>  <span class="at">cexCol =</span> <span class="fl">0.7</span>,</span>
<span id="cb21-9"><a href="#cb21-9"></a>  <span class="at">main =</span> <span class="st">"Heatmap.2"</span>,</span>
<span id="cb21-10"><a href="#cb21-10"></a>  <span class="at">trace =</span> <span class="st">"none"</span></span>
<span id="cb21-11"><a href="#cb21-11"></a>)</span>
<span id="cb21-12"><a href="#cb21-12"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="heatmap" class="slide level2">
<h2>Heatmap</h2>

<img data-src="./photos/Cluster_hm_zinc.png" class="r-stretch"></section>
<section id="advanced-excercise-1" class="slide level2">
<h2>Advanced Excercise 1</h2>
<p><br></p>
<p>How about the expression patterns of metal-related genes in control samples?</p>
</section>
<section id="advanced-excercise-2" class="slide level2">
<h2>Advanced Excercise 2</h2>
<p><br></p>
<p>Compare expression between control and zinc condition for halleri and lyrata side, respectively.</p>
</section>
<section id="homoeologous-ratio-test" class="slide level2">
<h2>6-3. Homoeologous Ratio Test</h2>
<p><br></p>
<ul>
<li>Calculate expression ratio using R script provided by “HomeoRoq”</li>
<li>Required files
<ul>
<li>RPKM matrix</li>
<li>label info</li>
<li>three R scripts</li>
</ul></li>
</ul>

<aside><div>
<p><a href="https://academic.oup.com/nar/article/42/6/e46/2437554">Akama et al.&nbsp;<em>Nucleic Acids Research</em> 2014</a></p>
</div></aside></section>
<section id="homoeologous-ratio-test-1" class="slide level2">
<h2>6-3. Homoeologous Ratio Test</h2>
<p><br></p>
<h3 id="label.txt">label.txt</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode numberSource t number-lines code-with-copy"><code class="sourceCode"><span id="cb22-1"><a href="#cb22-1"></a>"MUR_1_hal","MUR_1_lyr","MUR_R48h_1_hal","MUR_R48h_1_lyr"</span>
<span id="cb22-2"><a href="#cb22-2"></a>"MUR_2_hal","MUR_2_lyr","MUR_R48h_2_hal","MUR_R48h_2_lyr"</span>
<span id="cb22-3"><a href="#cb22-3"></a>"MUR_3_hal","MUR_3_lyr","MUR_R48h_3_hal","MUR_R48h_3_lyr"</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="calculate-homoeologous-ratio" class="slide level2">
<h2>6-3-1. Calculate homoeologous ratio</h2>
<p>!! Go back to terminal!</p>

<img data-src="./photos/start.png" class="r-stretch"></section>
<section id="calculate-homoeologous-ratio-1" class="slide level2">
<h2>6-3-1. Calculate homoeologous ratio</h2>
<p><br></p>
<h3 id="make-sure-you-are-in-the-scripts-directory.">Make sure you are in the <code>scripts</code> directory.</h3>
<div class="sourceCode" id="cb23"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1"></a><span class="ex">$</span> cd src/scripts</span>
<span id="cb23-2"><a href="#cb23-2"></a><span class="ex">scripts</span> $ R <span class="at">--vanilla</span> <span class="at">--slave</span> <span class="at">--args</span> ../exp/pval.txt ../exp/homoeolog_RPKM.txt label.txt <span class="op">&lt;</span> calcpval_one.R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="calculate-homoeologous-ratio-2" class="slide level2">
<h2>6-3-1. Calculate homoeologous ratio</h2>
<p><br></p>
<h3 id="make-sure-you-are-in-the-scripts-directory.-1">Make sure you are in the <code>scripts</code> directory.</h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1"></a><span class="ex">$</span> cd src/scripts</span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="ex">scripts</span> $ R <span class="at">--vanilla</span> <span class="at">--slave</span> <span class="at">--args</span> ../exp/pval.txt ../exp/homoeolog_RPKM.txt label.txt <span class="op">&lt;</span> calcpval_one.R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br></p>
<p>This process takes a while… How about a coffee break?</p>
</section>
<section id="plot-homoeolog-ratio" class="slide level2">
<h2>6-3-2. Plot homoeolog ratio</h2>
<p>Make sure you are in the <code>exp</code> directory!</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1"></a><span class="ex">scripts</span> $ cd ../exp</span>
<span id="cb25-2"><a href="#cb25-2"></a><span class="ex">exp</span> $ Rscript ../scripts/plot_homeoroq.R pval.txt MUR MUR_R48h</span>
<span id="cb25-3"><a href="#cb25-3"></a><span class="ex">exp</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="plot-homoeolog-ratio-1" class="slide level2">
<h2>6-3-2. Plot homoeolog ratio</h2>
<p>Make sure you are in the <code>exp</code> directory!</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1"></a><span class="ex">scripts</span> $ cd ../exp</span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="ex">exp</span> $ Rscript ../scripts/plot_homeoroq.R pval.txt MUR MUR_R48h</span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="ex">exp</span> $</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Now you got two pdf files.</p>
<ul>
<li>hist_overdispersion: Histogram of expression ratio</li>
<li>ratio_overdispersion: Scatterplot of the ratio of homeolog expression level between two conditions</li>
</ul>
</section>
<section id="histogram-of-expression-ratio" class="slide level2">
<h2>Histogram of expression ratio</h2>

<img data-src="./photos/hist_overdispersion_MUR_vs_MUR_48h.png" class="r-stretch"></section>
<section id="homoeolog-ratio-plots" class="slide level2">
<h2>Homoeolog ratio plots</h2>

<img data-src="./photos/ratio_overdispersion_MUR_vs_MUR_48h.png" class="r-stretch"><div class="footer footer-default">

</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="HomoeologousExpressionAnalysis_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>